build_tag_push.py

```diff
@@ -1,101 +1,130 @@
+import sys 
+import os
+import subprocess
+from shutil import copyfile # Allowing me to copy dockerfile context to new temp dir
+
+Docker_Hub_Name = "alyssakelley" # This is my personal docker namespace/username for testing.
+# If we want to make this a user input option --> Docker_Hub_Name = input('Enter your Docker namespace: ')
 
 # Step 1 - Read the app.yml file = COMPLETE.
-def parseAddFile(filename):
-	"""This is using the yaml built in to parse the file automatically with the file name that is opened. Loading info from a yaml file."""
+
+def parseAppFile(filename):
+	''' 
+	This function is using the yaml built in to parse the file which can be done by searching for 
+	the key in the dictionary for the yaml file. 
+
+	Example: name = dict_of_contents['name']
+	'''
 	with open(filename, "r") as file_contents:
 		dict_of_contents = yaml.load(file_contents) # might be sys.argv[1] with the GitHub URL to the yml file as the argument when running the python scrip
-	name = dict_of_contents['name']
-	#print name
-	build_script = dict_of_contents['build']
-	#print build_script
-	test_call = dict_of_contents['test']
-	#print test_call
-	docker_context_build = dict_of_contents['docker_context'][0]
-	#print docker_context_build
-	docker_context_static = dict_of_contents['docker_context'][1]
-	#print docker_context_static
-	return name, build_script, test_call, docker_context_build, docker_context_static # Returns a list of the values from the dictionary dict_of_contents
-
-# Step 2 - Builds and test the app according to that config file = NOT COMPLETE.
-# There needs to be a Build File .travis.yml on their github
-# Seems as though the GitHub repo needs to be added to the Travis CI account and then when there is a new push it will
-# run the tests in the .travis.yml file that is in that repo. Ask for clarification...
-#
-#
-#
-#
-#
-#
-#
-#
-#
-#
-
-# Step 3 - Runs docker build with the right context = PARTIALLY COMPLETE.
-# They will have their own Dockerfile in their GitHub repo (I am assuming) and this file is a text file with 
-# instructions to build image. Dockerfile -> docker build -> Docker Image, it is basically an automation of Docker Image
-# Creation. In the DockerFile, you generally do: FROM = tells Docker which image you base your image on, 
+	# name = dict_of_contents['name']
+	# build_script = dict_of_contents['build']
+	# test_call = dict_of_contents['test']
+	# docker_context_build = dict_of_contents['docker_context'][0]
+	# docker_context_static = dict_of_contents['docker_context'][1]
+	return dict_of_contents
+
+# Step 2 - Builds and test the app according to that config file = PENDING.
+
+# Is this something other than working with Travis CI or is this part of the next step? 
+# I was under the impression that Travis CI is what we are using to build and test their app. 
+# If not, can you elaborate more on what you want this step to be?
+
+def createContextDirectory(contextArray):
```

Comment:
by convention Python variables are "snake case" so `context_array` instead of `contextArray`.

There's a free tool called pep8 that checks your code to ensure it conforms to the standard Python style guide. You should start running it over your code. Do a `pip install pep8` to install it and then before you submit a PR run `pep8 build_tag_push.py` and fix any errors it spits out.